source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -178,16 +193,16 @@ namespace aspect
                    ExcMessage ("This initial condition can only be used if the geometry "
                                "is an ellipsoidal chunk."));
 
-
-      std::ifstream input1(isotherm_file_name.c_str());
+      const std::string data_file = data_directory+isotherm_file_name;
+      std::ifstream input1(data_file.c_str());
       AssertThrow (input1.is_open(),
-                   ExcMessage (std::string("Can't read from file <") + isotherm_file_name + ">"));
+                   ExcMessage (std::string("Can't read from file <") + data_file + ">"));
```

Comment:
We have recently replaced this kind of code by a utility function that handles reading a file more efficiently in parallel computations. Please take a look at 'source/particle/generator/ascii_file.cc' and replace the lines above by lines 37,38 from that file. You will also need to include 'aspect/utilities.h' at the beginning of the source file to use 'Utilities::read_and_distribute_file_content'. The reasoning behind this is explained in #761.